moduleframework/mtf_scheduler.py

```diff
@@ -0,0 +1,176 @@
+
+    # some useful bash variables
+    # there are more possible variables up to the doc, not sure what could be like options too
+    group = parser.add_argument_group('additional arguments are like environment variables up to the http://meta-test-family.readthedocs.io/en/latest/user_guide/environment_variables.html ')
+    group.add_argument("--module", action="store", help='Module type, like: docker, nspawn or rpm')
+    group.add_argument("--debug", action="store_true", help='passes AVOCADO_LOG_DEBUG=yes DEBUG=yes')
+    group.add_argument("--config", action="store", help='CONFIG=config.yaml')
+    group.add_argument("--url", action="store", help='URL overrides the value of module.docker.container or module.rpm.repo.')
+    group.add_argument("--modulemdurl", action="store", help='overwrites the location of a moduleMD file')
+
+    # unknown options are forwarded to avocado run
+    args, unknown = parser.parse_known_args()
+
+    common.print_debug("Options: linter={0}, setup={1}, action={2}, module={3}".format(
+        args.linter, args.setup, args.action, args.module))
+    common.print_debug("remaining options for avocado: {0}".format(unknown))
+
+    # environment usage:
+    #   read: os.environ.get('MODULE')
+    #   write: os.environ['MODULE'] 
+
+    # MODULE could be from:
+    #   1. common.get_module_type() ... it reads config.yaml and treceback if it doesn't exist
+    #   2. environment ... MODULE=docker etc
+    #   3. argument ... --module=docker
+    try:
+        args.module=common.get_module_type()
+        # TODO it wrongly writes: 'Using default minimal config ...', change in common.py
+    except moduleframework.mtfexceptions.ModuleFrameworkException:
+        pass
+
+    if os.environ.get('MODULE') is not None:
+        # environment is the highest priority because mtf uses environment (too much)
+        args.module=os.environ['MODULE']
+
+    if args.module in common.get_backend_list():
+        os.environ['MODULE']=args.module
+    else:
+        common.print_info("MODULE={0} ; we support {1} \n === expecting your magic, enjoy! === \n"
+                          "{2}".format(os.environ.get('MODULE'), common.get_backend_list(), parser.epilog))
+        # TODO: what to do here? whats the defaults value for MODULE, do I know it?
+        # exit(1)
+
+    common.print_debug("MODULE={0}".format(os.environ.get('MODULE')))
+
+    # uses additional arguments
+    if args.debug:
+        os.environ['AVOCADO_LOG_DEBUG']='yes'
+        os.environ['DEBUG']='yes'
+    if args.config:
+        os.environ['CONFIG']=args.config
+    if args.url:
+        os.environ['URL']=args.url
+    if args.modulemdurl:
+        os.environ['MODULEMDURL']=args.modulemdurl
+
+    return args, unknown
+
+class AvocadoStart(object):
+    def __init__(self, args, unknown):
+
+        self.module = os.environ['MODULE']
+        self.action = args.action
+
+        # its used for filepath in loadCli:
+        self.tmppath = tempfile.mktemp()
```

Comment:
this creates file everytime and in case you'll use another action like `list` it causes that it create file in constructor and does not remove this file anywhere, it is connected trouble with `def avocado_run` ,
name  and login in main invocate that it will be called just in case `avocado run` command, but it takes  `self.action` param, so that it is possible to invoke it with another action -> it causes troubles with this file, because `list` action does not allow to pass --json..* params or similar

Reply:
done